babel/core.py

```diff
@@ -191,10 +193,13 @@ def __init__(
         self.script = script
         #: the variant code
         self.variant = variant
+        #: the modifier
+        self.modifier = modifier
         self.__data = None
 
         identifier = str(self)
-        if not localedata.exists(identifier):
+        withoutmodifier = identifier.split('@', 1)[0]
```

Comment:
Can you move this to a property, e.g. `identifier_without_modifier`? I think we'd need to use this in the `_data` property too.

When you do that, can you also make it use `partition` instead:

```suggestion
        without_modifier = identifier.partition('@')[0]
```

Reply:
I can make this a property, sure, but I didn't, because `identifier` also isn't one.

I am unsure about `_data`. I didn't really dive this far on purpose because I didn't invest time trying to figure out where all the `*.dat` files are being created from.

As pertaining to the use of `partition`: I learnt something new. In this case, however, I don't think it matters much, but I'll happily change it.